assets/eip-5139/semver.md

```diff
@@ -0,0 +1,373 @@
+it may be best to perform a major version release, even though the fix could
+strictly be considered a patch release. Remember, Semantic Versioning is all
+about conveying meaning by how the version number changes. If these changes
+are important to your users, use the version number to inform them.
+
+### How should I handle deprecating functionality?
+
+Deprecating existing functionality is a normal part of software development and
+is often required to make forward progress. When you deprecate part of your
+public API, you should do two things: (1) update your documentation to let
+users know about the change, (2) issue a new minor release with the deprecation
+in place. Before you completely remove the functionality in a new major release
+there should be at least one minor release that contains the deprecation so
+that users can smoothly transition to the new API.
+
+### Does SemVer have a size limit on the version string?
+
+No, but use good judgment. A 255 character version string is probably overkill,
+for example. Also, specific systems may impose their own limits on the size of
+the string.
+
+### Is "v1.2.3" a semantic version?
+
+No, "v1.2.3" is not a semantic version. However, prefixing a semantic version
+with a "v" is a common way (in English) to indicate it is a version number.
+Abbreviating "version" as "v" is often seen with version control. Example:
+`git tag v1.2.3 -m "Release version 1.2.3"`, in which case "v1.2.3" is a tag
+name and the semantic version is "1.2.3".
+
+### Is there a suggested regular expression (RegEx) to check a SemVer string?
+
+There are two. One with named groups for those systems that support them
+(PCRE [Perl Compatible Regular Expressions, i.e. Perl, PHP and R], Python
+and Go).
+
+See: <https://regex101.com/r/Ly7O1x/3/>
+
+```
+^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+```
+
+And one with numbered capture groups instead (so cg1 = major, cg2 = minor,
+cg3 = patch, cg4 = prerelease and cg5 = buildmetadata) that is compatible
+with ECMA Script (JavaScript), PCRE (Perl Compatible Regular Expressions,
+i.e. Perl, PHP and R), Python and Go.
+
+See: <https://regex101.com/r/vkijKf/1/>
+
+```
+^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+```
+
+About
+-----
+
+The Semantic Versioning specification was originally authored by [Tom
+Preston-Werner](https://tom.preston-werner.com), inventor of Gravatar and
+cofounder of GitHub.
+
+If you'd like to leave feedback, please [open an issue on
+GitHub](https://github.com/semver/semver/issues).
+
+License
+-------
+
+[Creative Commons ― CC BY 3.0](https://creativecommons.org/licenses/by/3.0/)
```

Comment:
I'm generally against including anything that isn't CC0 in the repository.  It complicates things notably, as the repository's LICENSE file is CC0.  It feels like you can just mention "Semantic Versioning 2.0.0" and if someone wants, they can look up exactly what that means on the internet (where they are certain to find the answer).

Reply:
Directing someone to a standard by name is far worse than using an external link, and we already discourage external links. With a name, you get the same concerns about immutability and availability _plus_ the reader has to go and find the document themselves.

Since SemVer is an integral part of this spec, and it's freely redistributable, I feel pretty strongly I should be able to include it as an asset.

Reply:
I really don't have a problem linking externally to Semver. There are much worse external links. The assets here is fine by me as an alternative.

Reply:
I would take it a step further and say that *most* external links are worse than a link to semver.  However. Do you really want to take the time to debate the nuances of that very fuzzy line with every EIP author from now until the end of time?